compile error due to _InterlockedAdd() with Windows SDK >= 10.x

When compiling a project using Atomic.hpp with Visual Studio on 64-bit and Windows SDK >= 10.x you get the following compile error:

cat/threads/Atomic.hpp(205): error C3861: '_InterlockedAdd': identifier not found

As far as my understanding goes, _InterlockedAdd() is MS specific and only for ARM targets (see [1]). The fact that it compiles with Windows SDK < 10.x seems to be just by luck due to the fact how InterlockedAdd() is implemented in winnt.h.

Prior to WindowsSDK 10.x winnt.h defined InterlockedAdd as _InterlockedAdd. The following definition of InterlockedAdd() therefore defined _InterlockedAdd() in the end.
As of WindowsSDK 10.x this was changed and winnt.h now defines InterlockedAdd as _InlineInterelockedAdd. Hence _InterlockedAdd() will no longer be defined.

Also note that the "real" _InterlockedAdd() definition comes from intrins.h. This however is only defined for ARM (or in VS <= 2010 IPF) target platforms and was never defined for AMD64-bit targets.

[1] https://docs.microsoft.com/en-us/cpp/intrinsics/interlockedadd-intrinsic-functions

--- a/Atomic.hpp
+++ b/Atomic.hpp
@@ -316,7 +316,7 @@ u32 Atomic::Add(volatile u32 *x, s32 y)
 
 #if defined(CAT_COMPILER_MSVC) && defined(CAT_WORD_64)
 
-    u32 result = _InterlockedAdd((volatile LONG*)x, y) - y;
+    u32 result = InterlockedAdd((volatile LONG*)x, y) - y;
 
 	CAT_FENCE_COMPILER
 	return result;

